### PR TITLE
Add support for Flask 2.0

### PR DESCRIPTION
**Pull request by brettlangdon**
Necessary changes to support Flask 2.0

Main difference the removal of internal `_register_error_handler`.

I also removed testing on `~=0.10.0` and `~=0.11.0` since those are quite old at this point (0.11.x was 2016).

### CODE REVIEW

**Comment by brettlangdon** [comment]
CI is blocked on https://github.com/pallets/flask/pull/4049, or else we can only test with `>=1.1.0,<1.1.3`

**Comment by davidism** [comment]
We do not support Python < 3.6 any longer.

File: ddtrace/contrib/flask/patch.py [hunk 1]
```diff
@@ -429,7 +436,7 @@ def _wrap(template, context, app):
     return _wrap(*args, **kwargs)
 
 
-def traced_register_error_handler(wrapped, instance, args, kwargs):
+def traced__register_error_handler(wrapped, instance, args, kwargs):
     """Wrapper to trace all functions registered with flask.app.register_error_handler"""
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/flask/patch.py:440]:
```suggestion
    """Wrapper to trace all functions registered with flask.app._register_error_handler"""
```
?

**Review by Kyle-Verhoog — COMMENTED** [review]
Surprisingly not bad to deal with considering the major bump! Just one nit 🙂

**Review by jd — DISMISSED** [review]

File: riotfile.py [hunk 2]
```diff
@@ -654,6 +666,7 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
                 "python-memcached": latest,
                 "redis": "~=2.0",
                 "blinker": latest,
+                "werkzeug": "<2.0",
```
Comment (nizox) [inline riotfile.py:669]:
```suggestion
                "werkzeug": "<1.0",
```

Flask-Cache relies on `werkzeug.import_string` which was removed in 1.0

Comment (brettlangdon) [reply riotfile.py:669]: ah, good catch!

**Review by jd — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]

**Comment by brettlangdon** [comment]
@Mergifyio backport 0.49

**Pull request merged**